pipelines/shared/Jenkinsfile

```diff
@@ -249,14 +281,49 @@ pipeline {
                     intention.eventUrl = env.BUILD_URL
                     intention.eventProvider = env.EVENT_PROVIDER
                     intention.intention.actions[1].service.environment = 'production'
-                    intention.intention.actions[2].service.environment = 'production'
+                    intention.intention.actions[2].service.environment = 'production'                    
+                    intention.intention.actions[2].package.version = env.PODMAN_CD_VERSION
                     intention.open(NR_BROKER_JWT)
-                    intention.startAction("login")
+                    intention.startAction("login")                    
+                    def vaultToken = intention.provisionToken("login", CONFIG_ROLE_ID)
+                    def vault = new Vault(vaultToken)
+                    def cdCreds = vault.read('groups/data/appdelivery/jenkins-isss-cdua')
+                    env.PODMAN_CD_USER = cdCreds['username_lowercase']
+                    env.PODMAN_CD_PASS = cdCreds['password']
+                    def registryCreds = vault.read("${env.ARTIFACTORY_VAULT_PATH}")
```

Comment:
Should use vault.readToObject(). See: https://apps.nrs.gov.bc.ca/int/stash/projects/FLUENTBIT/repos/fluentbit-deploy/browse/pipelines/automation/Jenkinsfile#41